src/WalletLink.ts

```diff
@@ -185,10 +173,21 @@ export class WalletLink {
    * all potential stale state is cleared.
    */
   public disconnect(): void {
-    if (typeof window.walletLinkExtension !== "undefined") {
-      window.walletLinkExtension.close()
+    const extension = this.getCBWalletExtension()
+    if (extension) {
+      extension.close()
     } else {
       this._relay?.resetAndReload()
     }
   }
+
+  private getCBWalletExtension(): WalletLinkProvider | undefined {
+    return window.walletLinkExtension
```

Comment:
this will include the new variable added for rebranding and resolve whether extension is installed.
```suggestion
    return window.walletLinkExtension || window.coinbaseWallet
```